dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -126,7 +132,8 @@ HRESULT GetCLSIDForFileTypeFromExtension(LPWSTR pExtension, LPCWSTR KeyName, CLS
 
 HRESULT GetCLSIDForFileType(PCUIDLIST_RELATIVE pidl, LPCWSTR KeyName, CLSID* pclsid)
 {
-    LPWSTR pExtension = ExtensionFromPidl(pidl);
+    WCHAR buf[256];
```

Comment:
256 -> MAX_PATH
Here and for the other buffer below?

Reply:
It's not a path, its a filename. An extension is never going to be that long but theoretical maximum is '.' + 254 + '\0'.